samcli/lib/providers/sam_function_provider.py

```diff
@@ -78,23 +78,52 @@ def get(self, name: str) -> Optional[Function]:
         if not name:
             raise ValueError("Function name is required")
 
-        # support lookup by full_path
+        get_f = None
+
         if name in self.functions:
-            return self.functions.get(name)
+            # support lookup by full_path
+            get_f = self.functions.get(name)
+
+        if not get_f:
+            # If function is not found by full path, search through all functions
+
+            found_fs = []
+
+            for f in self.get_all():
+                if name in (f.function_id, f.name, f.functionname):
+                    found_fs.append(f)
+
+            # If multiple functions are found, only return one of them
+            if len(found_fs) > 1:
+                found_fs.sort(key=lambda f0: f0.full_path.lower())
+
+                LOG.warning(
+                    "Multiple functions found with keyword %s! Function %s will be invoked! "
+                    "If it's not the function you are going to invoke,"
+                    "please choose one of them from below:",
+                    name,
+                    found_fs[0].full_path,
+                )
+
+                for found_f in found_fs:
+                    LOG.warning(found_f.full_path)
 
-        for f in self.get_all():
-            if name in (f.name, f.functionname):
-                self._deprecate_notification(f.runtime)
-                return f
+                get_f = found_fs[0]
 
-        return None
+            elif len(found_fs) == 1:
+                get_f = found_fs[0]
+
+        if get_f:
+            self._deprecate_notification(get_f.runtime)
+
+        return get_f
 
     def _deprecate_notification(self, runtime: Optional[str]) -> None:
         if runtime in self._deprecated_runtimes:
             message = (
                 f"WARNING: {runtime} is no longer supported by AWS Lambda, "
                 "please update to a newer supported runtime. SAM CLI "
-                f"will drop support for all deprecated runtimes {self._deprecated_runtimes} on May 1st. "
+                f"will drop support for all deprecated runtimes {self._deprecated_runtimes} on May 1st, 2020. "
```

Comment:
The linked Github issue is closed. Not in this update, but we should probably remove this as it doesn't look like we support these anymore.

Reply:
Thanks. I removed this line, and only kept the warning, and the Github issue link.